{{cookiecutter.repo_name}}/.github/workflows/gh-ci.yaml

```diff
@@ -0,0 +1,258 @@
+name: GH Actions CI
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+concurrency:
+  # Specific group naming so CI is only cancelled
+  # within same PR or on merge to main
+  group: {{ "${{ github.ref }}-${{ github.head_ref }}-${{ github.workflow }}" }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  main_tests:
+    if: "github.repository == '{{ cookiecutter.github_username }}/{{ cookiecutter.repo_name }}'"
+    runs-on: {{ '${{ matrix.os }}' }}
+    strategy:
+        fail-fast: false
+        matrix:
+          os: [macOS-latest, ubuntu-latest, windows-latest]
+          python-version: ["3.7", "3.8", "3.9"]
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Build information
+      run: |
+        uname -a
+        df -h
+        ulimit -a
+
+{% if cookiecutter.dependency_source == 'Dependencies from pip only (no conda)' %}
+    - name: Set up Python {{ '${{ matrix.python-version }}' }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: {{ '${{ matrix.python-version }}' }}
+
+    - name: Install testing dependencies
+      shell: bash
```

Comment:
Does just bash work? IIRC, we have to use `bash -l` or something similar. Also, @IAlibay managed to define it as a default so isn't needed for each stanza.

Reply:
Yup, `bash -l` is the default for most steps. On Mac it picks up system Python 2.7 instead of whatever version we've installed, so we need just simple `bash` for a couple steps.